### ISSUE TITLE
resource_info sync missing repository_id parameter

### ISSUE THREAD

**Issue by mpl241** [theforeman/foreman-ansible-modules#1099]
##### SUMMARY
When using theforeman.foreman.resource_info to retrieve sync information (resource: sync), ansible-playbook returns error: 

fatal: [localhost]: FAILED! => {"ansible_facts": {"discovered_interpreter_python": "/usr/bin/python"}, "changed": false, "msg": "Failed to list resource: KeyError: \"missing param 'repository_id' in parameters\""}

##### ISSUE TYPE
 - Bug Report

##### ANSIBLE VERSION

```
ansible 2.9.15
  config file = .ansible.cfg
  configured module search path = [u'.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /bin/ansible
  python version = 2.7.5 (default, Aug 13 2020, 02:51:10) [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]
```

##### COLLECTION VERSION

```
$ grep version /usr/share/ansible/collections/ansible_collections/theforeman/foreman/MANIFEST.json 
  "version": "1.5.0",
```

##### KATELLO/FOREMAN VERSION

```
$ rpm -q tfm-rubygem-katello foreman
tfm-rubygem-katello-3.16.0.13-1.el7sat.noarch
foreman-2.1.2.21-1.el7sat.noarch
```

##### STEPS TO REPRODUCE

```yaml
- hosts: localhost
  gather_facts: no
  vars:
    server_url: https://satellite.example.com/
    username: admin
    password: REDACTED
    organization: "My Organization"
  tasks:
    - name: get info about sync
      theforeman.foreman.resource_info:
        username: "{{ username }}"
        password: "{{ password }}"
        server_url: "{{ server_url }}"
        organization: "{{ organization }}"
        resource: sync
```

##### EXPECTED RESULTS

I expect information about repository sync tasks.  Exact output unknown since I've never seen any output yet.

##### ACTUAL RESULTS

```
$ ansible-playbook get-cv-info.yml -vvvv --diff
ansible-playbook 2.9.15
  config file = .ansible.cfg
  configured module search path = [u'.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /bin/ansible-playbook
  python version = 2.7.5 (default, Aug 13 2020, 02:51:10) [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]
Using .ansible.cfg as config file
setting up inventory plugins
host_list declined parsing /etc/ansible/hosts as it did not pass its verify_file() method
script declined parsing /etc/ansible/hosts as it did not pass its verify_file() method
auto declined parsing /etc/ansible/hosts as it did not pass its verify_file() method
Set default localhost to localhost
Parsed /etc/ansible/hosts inventory source with ini plugin
Loading callback plugin default of type stdout, v2.0 from /usr/lib/python2.7/site-packages/ansible/plugins/callback/default.pyc

PLAYBOOK: get-cv-info.yml *********************************************************************************************************************************************************************************
Positional arguments: get-cv-info.yml
diff: True
become_method: sudo
inventory: (u'/etc/ansible/hosts',)
forks: 5
tags: (u'all',)
verbosity: 4
connection: smart
timeout: 10
1 plays in get-cv-info.yml

PLAY [localhost] ******************************************************************************************************************************************************************************************
META: ran handlers

TASK [get info about sync] ********************************************************************************************************************************************************************************
task path: foreman-ansible-modules/get-cv-info.yml:111
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: 
<localhost> EXEC /bin/sh -c 'echo ~ && sleep 0'
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo .ansible/tmp `"&& mkdir "` echo .ansible/tmp/ansible-tmp-1607979616.92-5706-26636356774087 `" && echo ansible-tmp-1607979616.92-5706-26636356774087="` echo .ansible/tmp/ansible-tmp-1607979616.92-5706-26636356774087 `" ) && sleep 0'
<localhost> Attempting python interpreter discovery
<localhost> EXEC /bin/sh -c 'echo PLATFORM; uname; echo FOUND; command -v '"'"'/usr/bin/python'"'"'; command -v '"'"'python3.7'"'"'; command -v '"'"'python3.6'"'"'; command -v '"'"'python3.5'"'"'; command -v '"'"'python2.7'"'"'; command -v '"'"'python2.6'"'"'; command -v '"'"'/usr/libexec/platform-python'"'"'; command -v '"'"'/usr/bin/python3'"'"'; command -v '"'"'python'"'"'; echo ENDFOUND && sleep 0'
<localhost> EXEC /bin/sh -c '/usr/bin/python && sleep 0'
Using module file /usr/share/ansible/collections/ansible_collections/theforeman/foreman/plugins/modules/resource_info.py
<localhost> PUT .ansible/tmp/ansible-local-5695nqwG_h/tmp_37q2J TO .ansible/tmp/ansible-tmp-1607979616.92-5706-26636356774087/AnsiballZ_resource_info.py
<localhost> EXEC /bin/sh -c 'chmod u+x .ansible/tmp/ansible-tmp-1607979616.92-5706-26636356774087/ .ansible/tmp/ansible-tmp-1607979616.92-5706-26636356774087/AnsiballZ_resource_info.py && sleep 0'
<localhost> EXEC /bin/sh -c '/usr/bin/python .ansible/tmp/ansible-tmp-1607979616.92-5706-26636356774087/AnsiballZ_resource_info.py && sleep 0'
<localhost> EXEC /bin/sh -c 'rm -f -r .ansible/tmp/ansible-tmp-1607979616.92-5706-26636356774087/ > /dev/null 2>&1 && sleep 0'
The full traceback is:
WARNING: The below traceback may *not* be related to the actual failure.
  File "/tmp/ansible_theforeman.foreman.resource_info_payload_7fijec/ansible_theforeman.foreman.resource_info_payload.zip/ansible_collections/theforeman/foreman/plugins/module_utils/foreman_helper.py", line 94, in inner
    return f(self, *args, **kwargs)
  File "/tmp/ansible_theforeman.foreman.resource_info_payload_7fijec/ansible_theforeman.foreman.resource_info_payload.zip/ansible_collections/theforeman/foreman/plugins/module_utils/foreman_helper.py", line 675, in list_resource
    return self._resource_call(resource, 'index', params)['results']
  File "/tmp/ansible_theforeman.foreman.resource_info_payload_7fijec/ansible_theforeman.foreman.resource_info_payload.zip/ansible_collections/theforeman/foreman/plugins/module_utils/foreman_helper.py", line 621, in _resource_call
    return self._resource(resource).call(*args, **kwargs)
  File "/tmp/ansible_theforeman.foreman.resource_info_payload_7fijec/ansible_theforeman.foreman.resource_info_payload.zip/ansible_collections/theforeman/foreman/plugins/module_utils/_apypie.py", line 864, in call
    return self.api.call(self.name, action, params, headers, options, data, files)
  File "/tmp/ansible_theforeman.foreman.resource_info_payload_7fijec/ansible_theforeman.foreman.resource_info_payload.zip/ansible_collections/theforeman/foreman/plugins/module_utils/_apypie.py", line 482, in call
    return self._call_action(action, params, headers, data, files)
  File "/tmp/ansible_theforeman.foreman.resource_info_payload_7fijec/ansible_theforeman.foreman.resource_info_payload.zip/ansible_collections/theforeman/foreman/plugins/module_utils/_apypie.py", line 492, in _call_action
    route.path_with_params(params),
  File "/tmp/ansible_theforeman.foreman.resource_info_payload_7fijec/ansible_theforeman.foreman.resource_info_payload.zip/ansible_collections/theforeman/foreman/plugins/module_utils/_apypie.py", line 906, in path_with_params
    raise KeyError("missing param '{}' in parameters".format(param))
fatal: [localhost]: FAILED! => {
    "ansible_facts": {
        "discovered_interpreter_python": "/usr/bin/python"
    }, 
    "changed": false, 
    "invocation": {
        "module_args": {
            "full_details": false, 
            "organization": "My Organization", 
            "params": null, 
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "resource": "sync", 
            "search": "", 
            "server_url": "https://satellite.example.com/", 
            "username": "admin", 
            "validate_certs": true
        }
    }, 
    "msg": "Failed to list resource: KeyError: \"missing param 'repository_id' in parameters\""
}

PLAY RECAP ************************************************************************************************************************************************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   

$ ansible-playbook get-cv-info.yml --check --diff

PLAY [localhost] ******************************************************************************************************************************************************************************************

TASK [get info about sync] ********************************************************************************************************************************************************************************
fatal: [localhost]: FAILED! => {"ansible_facts": {"discovered_interpreter_python": "/usr/bin/python"}, "changed": false, "msg": "Failed to list resource: KeyError: \"missing param 'repository_id' in parameters\""}

PLAY RECAP ************************************************************************************************************************************************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   
```

**Comment by evgeni** [comment]
You can provide the `repository_id` using the [`params` option of the `resource_info` module](https://theforeman.github.io/foreman-ansible-modules/develop/plugins/resource_info_module.html#parameter-params)

So smth like this:

```yaml
- hosts: localhost
  gather_facts: no
  vars:
    server_url: https://satellite.example.com/
    username: admin
    password: REDACTED
    organization: "My Organization"
  tasks:
    - name: get info about sync
      theforeman.foreman.resource_info:
        username: "{{ username }}"
        password: "{{ password }}"
        server_url: "{{ server_url }}"
        organization: "{{ organization }}"
        resource: sync
        params:
          repository_id: 1
```

**Comment by mpl241** [comment]
@evgeni,
  Thanks for the quick reply.  I was 100% sure I tried to put `repository_id` into `param`, but it didn't work.  The only documented parameters I found under `params` was `organization_id`.  In any case, I get another error when placing `repository_id` in `params`:

```
TASK [get info about sync] ********************************************************************************************************************************************************************************
task path: ~/foreman-ansible-modules/dev/get-cv-info.yml:111
<localhost> ESTABLISH LOCAL CONNECTION FOR USER:
<localhost> EXEC /bin/sh -c 'echo ~ && sleep 0'
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo .ansible/tmp `"&& mkdir "` echo .ansible/tmp/ansible-tmp-1608042214.07-14101-37856300391879 `" && echo ansible-tmp-1608042214.07-14101-37856300391879="` echo .ansible/tmp/ansible-tmp-1608042214.07-14101-37856300391879 `" ) && sleep 0'
<localhost> Attempting python interpreter discovery
<localhost> EXEC /bin/sh -c 'echo PLATFORM; uname; echo FOUND; command -v '"'"'/usr/bin/python'"'"'; command -v '"'"'python3.7'"'"'; command -v '"'"'python3.6'"'"'; command -v '"'"'python3.5'"'"'; command -v '"'"'python2.7'"'"'; command -v '"'"'python2.6'"'"'; command -v '"'"'/usr/libexec/platform-python'"'"'; command -v '"'"'/usr/bin/python3'"'"'; command -v '"'"'python'"'"'; echo ENDFOUND && sleep 0'
<localhost> EXEC /bin/sh -c '/usr/bin/python && sleep 0'
Using module file /usr/share/ansible/collections/ansible_collections/theforeman/foreman/plugins/modules/resource_info.py
<localhost> PUT .ansible/tmp/ansible-local-140853FPgdn/tmpgtOR82 TO .ansible/tmp/ansible-tmp-1608042214.07-14101-37856300391879/AnsiballZ_resource_info.py
<localhost> EXEC /bin/sh -c 'chmod u+x .ansible/tmp/ansible-tmp-1608042214.07-14101-37856300391879/ .ansible/tmp/ansible-tmp-1608042214.07-14101-37856300391879/AnsiballZ_resource_info.py && sleep 0'
<localhost> EXEC /bin/sh -c '/usr/bin/python .ansible/tmp/ansible-tmp-1608042214.07-14101-37856300391879/AnsiballZ_resource_info.py && sleep 0'
<localhost> EXEC /bin/sh -c 'rm -f -r .ansible/tmp/ansible-tmp-1608042214.07-14101-37856300391879/ > /dev/null 2>&1 && sleep 0'
The full traceback is:
WARNING: The below traceback may *not* be related to the actual failure.
  File "/tmp/ansible_theforeman.foreman.resource_info_payload_W443Ho/ansible_theforeman.foreman.resource_info_payload.zip/ansible_collections/theforeman/foreman/plugins/module_utils/foreman_helper.py", line 94, in inner
    return f(self, *args, **kwargs)
  File "/tmp/ansible_theforeman.foreman.resource_info_payload_W443Ho/ansible_theforeman.foreman.resource_info_payload.zip/ansible_collections/theforeman/foreman/plugins/module_utils/foreman_helper.py", line 675, in list_resource
    return self._resource_call(resource, 'index', params)['results']
fatal: [localhost]: FAILED! => {
    "ansible_facts": {
        "discovered_interpreter_python": "/usr/bin/python"
    }, 
    "changed": false, 
    "invocation": {
        "module_args": {
            "full_details": false, 
            "organization": "My Organization", 
            "params": {
                "repository_id": 1
            }, 
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "resource": "sync", 
            "search": "", 
            "server_url": "https://satellite.example.com/", 
            "username": "admin", 
            "validate_certs": true
        }
    }, 
    "msg": "Failed to list resource: KeyError: 'results'"
}
```
If I use a `resource_id` that does not exist, I get a different error:
```
TASK [get info about sync] ********************************************************************************************************************************************************************************
fatal: [localhost]: FAILED! => {"ansible_facts": {"discovered_interpreter_python": "/usr/bin/python"}, "changed": false, "error": {"displayMessage": "Couldn't find repository '1000'", "errors": ["Couldn't find repository '1000'"]}, "msg": "Failed to list resource: HTTPError: 404 Client Error: Not Found"}
```

  Thanks for your help.  I am still finding my way around this module.  If found myself recreating the automation I used with `cvmanager`  with this Ansible module.  It's been frustrating to say the least.  I hope to pull some useful `sync` information for a `repository` so I can conditionally `publish` a `content view`.

Regards
Matthew LeSieur

**Comment by evgeni** [comment]
Maybe the docs are miss-leading here, I'll have a look - you can put any param in there that the API expects.

The call with the right id seems to work correctly, but the module fails to parse the reponse. I'll have a look why.

**Comment by evgeni** [comment]
Could you try applying the following patch and showing me what the result looks like for you?

```patch
diff --git plugins/module_utils/foreman_helper.py plugins/module_utils/foreman_helper.py
index 29dbd00c..c51a2411 100644
--- plugins/module_utils/foreman_helper.py
+++ plugins/module_utils/foreman_helper.py
@@ -682,7 +682,11 @@ class ForemanAnsibleModule(AnsibleModule):
 
         params = self._resource_prepare_params(resource, 'index', params)
 
-        return self._resource_call(resource, 'index', params)['results']
+        result = self._resource_call(resource, 'index', params)
+        if 'results' in result:
+            return result['results']
+        else:
+            return result
 
     def find_resource(self, resource, search, params=None, failsafe=False, thin=None):
         list_params = {}
```

**Comment by mpl241** [comment]
@evgeni,
  Patch applied:
```
# patch -p0 < ~/foreman-ansible-modules/theforeman-foreman-ansible-module.patch
patching file plugins/module_utils/foreman_helper.py
Hunk #1 succeeded at 672 (offset -10 lines).
```
No error this time, but the variable registered with the task is empty.
```
ok: [localhost] => {
    "ansible_facts": {
        "discovered_interpreter_python": "/usr/bin/python"
    }, 
    "changed": false, 
    "invocation": {
        "module_args": {
            "full_details": false, 
            "organization": "My Organization", 
            "params": {
                "repository_id": 1
            }, 
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "resource": "sync", 
            "search": "", 
            "server_url": "https://satellite.example.com/", 
            "username": "admin", 
            "validate_certs": true
        }
    }, 
    "resources": {}
}
```

**Comment by evgeni** [comment]
Thanks for trying. Yeah, I feared that. Had the same result in my env, but wasn't sure if that's the API or my rather broken env.

You could try looking at the `repositories` resource (and passing `id: ...` in params), that also has details about the last sync?

**Comment by evgeni** [comment]
> I am still finding my way around this module. If found myself recreating the automation I used with `cvmanager` with this Ansible module. It's been frustrating to say the least.

Anything we can do to make it less frustrating? Where (besides this issue) were your problems?

> I hope to pull some useful `sync` information for a `repository` so I can conditionally `publish` a `content view`.

The real problem is (and also was in cvmanager) that a synced repository does not imply changed content. cvmanager has some hacky code that tries to find the sync task and then parse the output of that, but it was so clumsy (and not reliable) that we decided not to port that over to FAM.

**Comment by mdellweg** [comment]
> Anything we can do to make it less frustrating? Where (besides this issue) were your problems?

If the information is there, but not accessible, we could try to provide filters to compile interesting information.

**Comment by mpl241** [comment]
@evgeni ,
  Actually, yes, I am able to get package count from `repositories`, so I don't think I need any `sync` info.  It looks like I can also get package counts from `content_view_versions` too, so I think I have what I need to perform conditional `publishes`.

  Regarding the documentation referencing the API, I was not sure what API I should be looking at.  Maybe being more explicit about what API would be good.  I figured out I needed to look at the Katello API and then found the possible `params` for `repositories` .  `search`ing `repositories` returned an empty `resources`, but using `params: name:` worked.  But, I didn't get the result I expected using `params: id:` though (bug?).  Using `params: name` worked as expected though.

Regards
Matthew LeSieur

**Comment by mpl241** [comment]
@evgeni 
  The code in cvmanager for finding repository changes worked well in my environment even if it was "hacky".  The alternatives were manually publishing content views when changes are made or always publishing every content view regardless if a change is made.  I was fortunate the hacky code in cvmanager allowed us to only publish when necessary.

  What is frustrating is having to reimplement cvmanager using the theforeman-forman Ansible module.  The module provides the interface, but not the logic that cvmanager handled.  Now I need to write a playbook/role to handle the logic that cvmanager provided.  So far, I can conditionally publish composite content views when a component changes, but I cannot conditionally publish a content view when a repository changes.  Foreman/Satellite can auto-publish a composite content view when a component content view changes, but content views do not have an auto-publish feature when a repository changes.  I am now trying to work out the Ansible logic to detect changes to repositories so content views can be published.

Regards
Matthew LeSieur

**Comment by mpl241** [comment]
@evgeni , @mdellweg 
  The logic I thought I can use to detect updates to repositories does not work because content view counts (package counts, errata counts, etc) are not accurate and may not match the repository counts.  I am looking for other ways to detect repository changes, but at this point, I must decide between manually publishing content views when there are changes or always publish content views.  Neither is great for a fully automated system.

Regards
Matthew LeSieur

**Comment by evgeni** [comment]
The code in cvmanager that decides whether a CV needs publishing is here: https://github.com/RedHatSatellite/katello-cvmanager/blob/847da1d375363411457371dfa0ba7321f747552f/cvmanager#L391-L478 -- almost 100 lines of code ;)

Let's dig into that.

First, it tries to find the timestamp a given CV was last published, this is done as follows:
1. look at the latest published version and take it's `published` attribute
2. if for some reason that wasn't found, take the `last_published` of the CV itself
3. if that also wasn't found, fall back to `0`

Next, find when the repos that are in the CV were last changed:
* for each repository in the CV, look at the `last_sync` attribute and compare it to the CV publish date
* if any repo is found that is "newer" than the CV, a publish will be scheduled
* unless the advanced `checkrepos` flag is set, which triggers more checks, but I'd prefer to avoid for now

If we ignore `checkrepos`, the above works, but has the drawback that it will publish every time a repo was synced, regardless whether there was an actual change or not in the repo. So if you have a Sync Plan or some other means of regular repository synchronization, the whole logic is moot and you can just publish every time you know the repo was synced.

I tried to implement that (and doing so required the patch from #1115), which resulted in the following playbook:

```yaml
- hosts: localhost
  gather_facts: false
  vars:
    foreman_username: admin
    foreman_password: changeme
    foreman_url: https://foreman.example.com
    foreman_validate_certs: False
    foreman_initial_organization: Default Organization
  tasks:
    - name: get info
      theforeman.foreman.resource_info:
        username: "{{ foreman_username }}"
        password: "{{ foreman_password }}"
        server_url: "{{ foreman_url }}"
        validate_certs: "{{ foreman_validate_certs }}"
        organization: "{{ foreman_initial_organization }}"
        search: 'name = RHEL7'
        resource: content_views
      register: cvs
    - debug:
        msg: "CV last published: {{ cvs.resources[0]['last_published'] }}"
    - name: get info
      theforeman.foreman.resource_info:
        username: "{{ foreman_username }}"
        password: "{{ foreman_password }}"
        server_url: "{{ foreman_url }}"
        validate_certs: "{{ foreman_validate_certs }}"
        organization: "{{ foreman_initial_organization }}"
        resource: repositories
        params:
          id: "{{ item }}"
      register: repos
      loop: "{{ cvs.resources[0]['repository_ids'] }}"
    - debug:
        msg: "Repo last synced: {{ item.resources[0]['last_sync'] }}"
      loop: "{{ repos.results }}"
```

But interestingly, all those `last_sync` params are `null` :(

**Comment by mpl241** [comment]
@evgeni
  I have logic in my playbook/role for comparing the last `repository` update and the last `composite content view` update.  This is an important check because I have the same `component` defined in multiple `composite content views`.  I don't want to trigger a `publish` every time I check a `composite content view`.  You can get the `last_sync.updated_at` variable by retrieving the list of `repositories` from the `content view version`.  I don't know why you can't get it directly from the `composite content view`.  Maybe it's an API bug.  Below is the code that works for me.  `last_check` is not defined when a `repository` is not synchronized.  Specifically, I have a `content view` that I only manually add packages to, so it will never have `last_check` defined.

```
- hosts: localhost
  gather_facts: false
  vars:
    foreman_username: "{{ username }}"
    foreman_password: "{{ password }}"
    foreman_url: "{{ server_url }}"
    foreman_validate_certs: True
    foreman_initial_organization: "{{ organization }}"
    ccv: "RHEL7"
  tasks:
    - name: get cv info
      theforeman.foreman.resource_info:
        username: "{{ foreman_username }}"
        password: "{{ foreman_password }}"
        server_url: "{{ foreman_url }}"
        validate_certs: "{{ foreman_validate_certs }}"
        organization: "{{ foreman_initial_organization }}"
        params:
          name: "{{ ccv }}"
        resource: content_views
      register: cvs
    - debug:
        msg: "CV last published: {{ cvs.resources[0]['last_published'] }}"
    - name: get cvv info
      theforeman.foreman.resource_info:
        username: "{{ foreman_username }}"
        password: "{{ foreman_password }}"
        server_url: "{{ foreman_url }}"
        validate_certs: "{{ foreman_validate_certs }}"
        organization: "{{ foreman_initial_organization }}"
        resource: content_view_versions
        search: 'content_view_id = {{ cvs.resources[0].id }}'
        params:
          environment_id: 1
      register: cvvs
    - debug:
        msg: "CVV last published: {{ cvvs.resources[0]['updated_at'] }}
    - name: get repo info
      theforeman.foreman.resource_info:
        username: "{{ foreman_username }}"
        password: "{{ foreman_password }}"
        server_url: "{{ foreman_url }}"
        validate_certs: "{{ foreman_validate_certs }}"
        organization: "{{ foreman_initial_organization }}"
        resource: repositories
        params:
          content_view_version_id: "{{ cvvs.resources[0].id | int }}"
          library: 1
      register: repos
    - debug:
        msg: "Repo last synced: {{ item.last_sync.ended_at }}"
      loop: "{{ repos.resources }}"
      loop_control:
        label: "{{ item.name }}"
      when: item.last_sync.ended_at is defined
```

**Comment by mpl241** [comment]
@evgeni 
  To take it a step further, I use a counter to keep track of how many `repositories` have a `last_sync` time after the content view version `updated_at` time.
```
- name: check if any repository for content view has been updated since the last content view update (skip if repository has never synced)
  set_fact:
    repo_newer: "{{ repo_newer | int + 1 }}"
  loop: "{{ repos_info.resources }}"
  loop_control:
    label: "{{ repos.name }}"
    loop_var: repo_item
  when: ( repo_item.last_sync.ended_at is defined ) and ( repo_item.last_sync.ended_at | regex_replace(' UTC$') | to_datetime() > cvv_info.updated_at | regex_replace(' UTC$') | to_datetime() )
```
  I then check the `repo_newer` variable when running the task that `publishes` the `composite content view`.

Regards
Matthew LeSieur

**Comment by evgeni** [comment]
I have opened https://projects.theforeman.org/issues/31590 to get the data from the source. Won't help today, but at least in the future :)

**Comment by mpl241** [comment]
@evgeni 
  Excellent, thanks!  Having Foreman indicate when a `repository` last changed verses the last sync would be ideal.  Unfortunately, I am using Satellite (6.8), so it may be months (years?) before I ever see the change.  It is a step in the right direction though.

  In the mean time, I am using a hacky workaround to my problem.  I use `cvmanager` to only `publish` repositories when a change is found then use an Ansible playbook/role I created to do all of the other heavy lifting.  Hopefully the `publish` functionality of `cvmanager` will continue to work until the `changed` flag is available for a `repository`.

Regards
Matthew LeSieur

**Comment by evgeni** [comment]
Yeah, I don't expect cvmanager to *break* anytime soon. (The only known issue is with CCVs that have their components set to "latest", but that is buggy since Satellite 6.3 or whenever that feature was introduced).

I just don't want to maintain it *if* anything happens ;)

**Comment by mpl241** [comment]
I started working on the Ansible Playbook/Role for managing Satellite content because of some errors thrown by `cvmanager`.  After a few quick searches, I assumed (incorrectly) there was an API incompatibility between Satellite and `cvmanager` after I upgraded to Satellite 6.8.  I checked `cvmanager` again for errors, and it appears to be working properly now.  I did fix a Satellite bug related to `python-nectar` the other day.  It is hard to say if it fixed the `cvmanager` error.  Unfortunately, I have no way of troubleshooting the errors, but I wanted to mention it.

https://bugzilla.redhat.com/show_bug.cgi?id=1893561
